src/path/sentry_path_windows.c

```diff
@@ -205,6 +220,25 @@ sentry__path_from_str(const char *s)
     return rv;
 }
 
+sentry_path_t *
+sentry__path_from_str_n(const char *s, size_t s_len)
+{
+    if (!s) {
+        return NULL;
+    }
+    sentry_path_t *rv = SENTRY_MAKE(sentry_path_t);
+    if (!rv) {
+        return NULL;
+    }
+    rv->path = sentry_malloc(sizeof(wchar_t) * (s_len + 1));
+    if (!rv->path) {
+        sentry_free(rv);
+        return NULL;
+    }
+    MultiByteToWideChar(CP_ACP, 0, s, -1, rv->path, (int)s_len + 1);
```

Comment:
You are right; there are a couple of things wrong:

* We need explicit size (excluding 0) analog to explicit length
* The length of the converted string includes the null character (which only "works" if there was a zero immediately following the string slice)
* The destination string must be explicitly null-terminated (if we don't have a _correctly sized_ null-terminated input string because it will find 0 at some point).
* In case of an error in `MultiByteToWideChar`, we don't free `rv`

I also rewrote the null-terminated version in terms of the ptr/len version.